tpch: improve documentation

rjnn opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/loadgen/44)
<!-- Reviewable:end -->

Commit:
tpch: improve documentation

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> w/ comments

Also, an example with SSL would be useful. It's not obvious what the preferred way of connecting to a secure cluster is.

---

Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[tpch/README, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMUbDHB8tRirJgB63:-KhTMUbDHB8tRirJgB64:byt6h5e) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L37)):*
> ```
> 1. Acquire a cockroach backup file, which will load into a cluster
> within minutes.
> 1a. Load the cockroach backup using the Enterprise backup/restore feature from the local filesystem.
> ```

Is this option worth mentioning? It seems to have significant caveats.

---

*[tpch/README, line 46 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMcKtCCAr-yMzoML6:-KhTMcKtCCAr-yMzoML7:b2t2oym) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L46)):*
> ```
> (1b) is the fastest and the recommended way - but requires access to
> Cockroach's Azure blob store. Ask someone at Cockroach Labs for the
> link, as we do pay Azure data egress fees on each download (not much,
> ```

Do we pay even when accessing from an Azure VM?

---

*[tpch/README, line 55 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMmuqbC45tvVxFmGZ:-KhTMmuqbC45tvVxFmG_:bcgiy8l) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L55)):*
> ```
> Followed by
> 
>     restore tpch.* from 'azure://backup-test/benchmarks/tpch/scalefactor-1?AZURE_ACCOUNT_NAME=cockroachbackuptest&AZURE_ACCOUNT_KEY=<ommitted>';
> ```

s/ommitted/omitted

---

*[tpch/README, line 84 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTNS6zNJt4rY7UV8sn:-KhTNS6zNJt4rY7UV8so:b-v3rh08) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L84)):*
> ```
> 
> If you put the .tbl files inside a directory named `data` in the same
> directory as the `tpch` binary, running ./tpch -load will load the
> ```

backticks around `./tpch -load`

---

*[tpch/README, line 85 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTNqS_S3bjT4axp6cx:-KhTNqS_S3bjT4axp6cy:b9jsipk) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L85)):*
> ```
> If you put the .tbl files inside a directory named `data` in the same
> directory as the `tpch` binary, running ./tpch -load will load the
> data into Cockroach. This currently takes a lot of time (~5 hours on a
> ```

I believe this takes < 1 hour now. Still longer doing a RESTORE though. This doc might age better if you say that this takes several times as long as a RESTORE.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-:-KhTO7lyKH3sDJPyeUgy:by8l6zl)*
<!-- Sent from Reviewable.io -->

rjnn commented:
Added a 'Using the load generator' section that provides example strings, including with and without SSL.

---

Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[tpch/README, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMUbDHB8tRirJgB63:-KhTTrV7N_glUQmesa_U:b-32hqxs) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L37)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

Is this option worth mentioning? It seems to have significant caveats.
</blockquote></details>

It's only useful for single node clusters. Updated and relegated to the last option.

---

*[tpch/README, line 46 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMcKtCCAr-yMzoML6:-KhTTCNOv17fpPScFl1_:b-h15n2t) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L46)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

Do we pay even when accessing from an Azure VM?
</blockquote></details>

We do not, but this excuse is for why we don't publish the link in the repository in the open. The fees really don't matter (they are pennies per download). Clarified.

---

*[tpch/README, line 55 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTMmuqbC45tvVxFmGZ:-KhTTQAL5_dFHX0lrtsZ:b-896fix) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L55)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

s/ommitted/omitted
</blockquote></details>

Done.

---

*[tpch/README, line 84 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTNS6zNJt4rY7UV8sn:-KhTTTEY9R7Ry9ZW87E9:b-896fix) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L84)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

backticks around `./tpch -load`
</blockquote></details>

Done.

---

*[tpch/README, line 85 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-KhTNqS_S3bjT4axp6cx:-KhTTZ4l8zR6bZj7aoPN:b-896fix) ([raw file](https://github.com/cockroachdb/loadgen/blob/5a43b5827e5d802e0a5d12ac474bb4b2033f2b54/tpch/README#L85)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

I believe this takes < 1 hour now. Still longer doing a RESTORE though. This doc might age better if you say that this takes several times as long as a RESTORE.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/44#-:-KhTThzQ1PL8JbxQZr1f:blttpwy)*
<!-- Sent from Reviewable.io -->
